src/Connection.cpp

```diff
@@ -121,8 +131,38 @@ int Connection::connect() {
   return m_alive ? 0 : -1;
 }
 
-int Connection::connectPoll(int fd, struct addrinfo* ai_ptr) {
-  int conn_rv = ::connect(fd, ai_ptr->ai_addr, ai_ptr->ai_addrlen);
+inline int Connection::local() {
+  int fd, flags, opt_keepalive = 1;
+
+  if ((fd = socket(AF_UNIX, SOCK_STREAM, 0)) == -1) {
+    log_err("socket()");
+    return -1;
+  }
+
+  if ((flags = fcntl(fd, F_GETFL, 0)) < 0 ||
+      fcntl(fd, F_SETFL, flags | O_NONBLOCK) < 0) {
+      log_err("setting O_NONBLOCK");
+      ::close(fd);
+      return -1;
+  }
+
+  setsockopt(fd, SOL_SOCKET, SO_KEEPALIVE, (void *)&opt_keepalive, sizeof opt_keepalive);
```

Comment:
I'm not actually sure whether this flag does anything on the client side, I just pulled it from

https://github.com/memcached/memcached/blob/9723c0ea8ec1237b8364410ba982af8ea020a2b6/memcached.c#L3898

Reply:
Oh, I forgot that I cross-referenced with the TCP connection code

https://github.com/douban/libmc/blob/fbf4acfed257ee64332da015661e945ea45793f8/src/Connection.cpp#L102